addons/Omen/Omen.lua

```diff
@@ -0,0 +1,259 @@
+
+good_col = "\\cs("..tostring(settings.good_R)..","..tostring(settings.good_G)..","..tostring(settings.good_B)..")"
+bad_col = "\\cs("..tostring(settings.bad_R)..","..tostring(settings.bad_G)..","..tostring(settings.bad_B)..")"
+omens = 0
+obj_time = 0
+floor_obj = "Waiting for objectives..."
+floor_clear = ""
+--image = texts.new("image", settings)
+image = texts.new("image")
+
+texts.color(image,settings.text_R,settings.text_G,settings.text_B)
+texts.size(image,settings.font_size)
+texts.pos_x(image,settings.pos_x)
+texts.pos_y(image,settings.pos_y)
+texts.bg_alpha(image,settings.bg_alpha)
+
+function reset_objectives()
+    objectives = {
+    [1] = {id=1,mes=0,amt=0,req=0},
+    [2] = {id=2,mes=0,amt=0,req=0},
+    [3] = {id=3,mes=0,amt=0,req=0},
+    [4] = {id=4,mes=0,amt=0,req=0},
+    [5] = {id=5,mes=0,amt=0,req=0},
+    [6] = {id=6,mes=0,amt=0,req=0},
+    [7] = {id=7,mes=0,amt=0,req=0},
+    [8] = {id=8,mes=0,amt=0,req=0},
+    [9] = {id=9,mes=0,amt=0,req=0},
+    [10] = {id=10,mes=0,amt=0,req=0}
+    }
+    obj_time = 0
+    floor_clear = ""
+end
+reset_objectives()
+
+function refresh()
+    header = floor_clear..floor_obj.."\\cr     Omens: "..omens
+    body = "\n Bonus Objectives    "..os.date('%M:%S', obj_time)
+    for k,v in pairs (hide_timer) do
+        if string.find(header,v) then
+            body = ""
+            texts.text(image,header)
+            return
+        end
+    end
+    for v, objective in ipairs(objectives) do
+        if objective.mes ~= 0 then
+            local msg = objective.mes
+            local cur = objective.amt
+            local fin = objective.req
+            if cur == fin then
+                body = body.."\n "..good_col..v..": "..messages[msg].short.." ["..cur.."/"..fin.."]\\cr"
+            elseif obj_time < 1 and cur < fin then
+                body = body.."\n "..bad_col..v..": "..messages[msg].short.." ["..cur.."/"..fin.."]\\cr"
+            else
+                body = body.."\n "..v..": "..messages[msg].short.." ["..cur.."/"..fin.."]"
+            end
+        end
+    end
+    body = string.gsub(body,"%-1","%?%?%?")
+    texts.text(image,header..body)
+end
+
+hide_timer = {"Kin","Gin","Kei","Kyou","Fu","Ou","Craver","Gorger","Thinker","Treasure","Waiting"}
+refresh()
+
+windower.register_event('prerender', function()
+    if obj_time < 1 then return end
+    if obj_time ~= (end_time - os.time()) then
+        obj_time = end_time - os.time()
+        refresh()
+    end
+end)
+
+windower.register_event('zone change', function(zone)
+    image:hide()
+    floor_obj = "Waiting for objectives..."
+    reset_objectives()
+    if zone == 292 then -- Reisenjima Henge
+        image:show()
+    end
+end)
+
+image:hide()
+if windower.ffxi.get_info().zone == 292 then -- 292 is the code for Reisenjima Henge
+    image:show()
+end
+
+windower.register_event('incoming text', function(original, modified, mode)
+	local objective = objectives[tonumber(original:match("^%d"))]
```

Comment:
Yeah, tested it and it works. Since I didn't 100% trust my custom test function, I went in and tested in real omen, seems to be working. Thanks!